text/0722-align-mount-route-apis.md

```diff
@@ -0,0 +1,73 @@
+---
+Stage: Accepted
+Start Date: 2021-02-12
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js
+RFC PR: https://github.com/emberjs/rfcs/pull/722
+---
+
+<!---
+Directions for above:
+
+Stage: Leave as is
+Start Date: Fill in with today's date, YYYY-MM-DD
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies
+RFC PR: Fill this in with the URL for the Proposal RFC PR
+-->
+
+# Align mount and route APIs
+
+## Summary
+
+Update the Ember `mount` API to accept a function argument which defines the Engine’s routes similar to how the `route` API works today.
+
+## Motivation
+
+Some Ember applications utilize in-repo ember-engines strictly for their lazy loading and code organization features and not for their isolation features. In this context the isolation actually causes some maintenance overhead and prevents build optimization.
+
+If isolation is not a requirement and all engines in an application are in-repo it makes more sense to define all of the routes at the application level in the router.js.
```

Comment:
Could you please clarify more the purpose of update the Ember `mount` API here? I'm trying to understand the reason for that. Maybe put some use case can make more sense.